src/consensus/encode.rs

```diff
@@ -139,9 +139,11 @@ impl From<psbt::Error> for Error {
 
 /// Encode an object into a vector
 pub fn serialize<T: Encodable + ?Sized>(data: &T) -> Vec<u8> {
-    let mut encoder = Vec::new();
+    let serialized_len = data.serialized_len();
+    let mut encoder = Vec::with_capacity(serialized_len);
     let len = data.consensus_encode(&mut encoder).expect("in-memory writers don't error");
     debug_assert_eq!(len, encoder.len());
+    debug_assert_eq!(serialized_len, encoder.len());
```

Comment:
This implicitly unit tests the implementations of `serialize_len()` as long as we serialize somewhere in unit tests, right? I had a look at `consensus/encode.rs` and there are a bunch of tests but I don't know if we have full coverage. Did you do this on purpose? If so, perhaps when the final version goes up could you add some text justifying not adding explicit unit tests for `serialize_len` please so other reviewers don't wonder why there is none added?

Reply:
> This implicitly unit tests the implementations of serialize_len() as long as we serialize somewhere in unit tests, right?

Yes, that was the idea to have a quick check, I will add the comment and also some explicit unit tests in the final version.